client/src/app/pages/migration-targets/migration-targets.tsx

```diff
@@ -124,22 +134,39 @@ export const MigrationTargets: React.FC = () => {
           response.data.id,
         ]);
       }
-    }
 
+      // Make sure the new target's provider is part of the provider's filter so it can be seen
+      if (filterState.filterValues["provider"]) {
+        let newProviderFilter: FilterValue = null;
+
+        const targetProvider = response.data.provider;
+        if (targetProvider) {
+          const fv = filterState.filterValues["provider"];
+          newProviderFilter = fv.includes(targetProvider)
+            ? fv
+            : [...fv, targetProvider];
+        }
+
+        filterState.setFilterValues({
+          name: filterState.filterValues["name"],
```

Comment:
can we just unpack all the filters to be future proof ? i.e. `{... filterState.filterValues, provider: newProviderFilter}`